core/src/test/java/org/apache/calcite/tools/PlannerTest.java

```diff
@@ -1525,4 +1527,67 @@ private void checkView(String sql, Matcher<String> matcher)
     final RelRoot root = planner.rel(validate);
     assertThat(toString(root.rel), matcher);
   }
+
+  /** Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-4642">[CALCITE-4642]
+   * Checks that custom type systems can be registered in a planner by
+   * comparing options for converting unions of chars.</a>.
+   */
+  @Test void testCustomTypeSystem() throws Exception {
+    final String sql = "select Case when DEPTNO <> 30 then 'hi' else 'world' end from dept";
+    final String expectedVarying = "LogicalProject("
+        + "EXPR$0=["
+        + "CASE(<>($0, 30),"
+        + " 'hi':VARCHAR(5), "
+        + "'world':VARCHAR(5))])\n"
+        + "  LogicalValues("
+        + "tuples=[[{ 10, 'Sales' },"
+        + " { 20, 'Marketing' },"
+        + " { 30, 'Engineering' },"
+        + " { 40, 'Empty' }]])\n";
```

Comment:
```suggestion
    final String expectedVarying = ""
        + "LogicalProject(EXPR$0=[CASE(<>($0, 30), 'hi':VARCHAR(5), 'world':VARCHAR(5))])\n"
        + "  LogicalValues(tuples=[[{ 10, 'Sales' }, { 20, 'Marketing' }, { 30, 'Engineering' }, { 40, 'Empty' }]])\n";
```

Reply:
I'm in favor of inserting line-breaks by hand to keep lines short. I think the previous is fine.

Reply:
Ok, whatever. It took me a while to understand the plan, as I had to search for `\n`, so I asked.

Reply:
Yeah, that's valid. It's difficult to see the  relational operators 'LogicalProject' and 'LogicalValues'. Sometimes I ensure that continuation lines have no preceding spaces at, which visually distinguishes them from the indented relational operators. Like this:

```
"LogicalProject(EXPR$0=[CASE(<>($0, 30), "
        + "'hi   ', "
        + "'world')])\n"
        + "  LogicalValues("
        + "tuples=[[{ 10, 'Sales      ' }, "
        + "{ 20, 'Marketing  ' }, "
        + "{ 30, 'Engineering' }, "
        + "{ 40, 'Empty      ' }]])\n";
```

It works well in large, very-indented plans.

Reply:
Exactly. I do see the reason to wrap the plan line after 200-300 chars, however, the current case is not worth the discussion.
I think non-wrapped is readable, and the wraps make it harder to read.

However, I do not object, and the rest in the patch looks great.